sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```
@@ -330,4 +336,134 @@ public class ChatClient {
+            completionHandler(.failure(AzureError.client("Invalid URL.", error)))
+        }
+    }
+
+
+    /// Stop push notifications.
+    /// - Parameter completionHandler: Success indicates push notifications have been stopped.
+    public func stopPushNotifications(
+        completionHandler: @escaping (Result<HTTPResponse?, AzureError>) -> Void
+    ) {
+        guard pushNotificationsStarted else {
+            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")))
+            return
+        }
+
+        guard let registrarClient = registrarClient else {
+            completionHandler(.failure(AzureError.client("Failed to stop push notifications. There is no registrarClient.")))
+            return
+        }
+        
+        registrarClient.deleteRegistration { result in
+            switch result {
+            case .success(let response):
+                self.pushNotificationsStarted = false
+                completionHandler(.success(response))
+            case .failure(let error):
+                self.options.logger.error("Failed to stop push notifications with error: \(error.localizedDescription)")
+                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)))
+            }
+        }
+    }
+
+    /// Handle the data payload for an incoming push notification.
+    /// - Parameters:
+    ///   - notification: The APNS push notification.
+    ///   - completionHandler: Receives the PushNotificationEvent from the push notification payload.
+    public func handlePush(
+        notification: [AnyHashable : Any],
+        completionHandler: PushNotificationEventHandler
+    ) {
+        guard let payload = notification["data"] as? [String: AnyObject] else {
+            options.logger.error("Push notification does not contain data payload.")
+            return
```

This represents a silent failure. You aren't calling completionHandler, like you do below in the failure case.

---

Fixed and will push the code change later.